packages/contracts/contracts/plugin/PluginSetupProcessor.sol

```diff
@@ -87,87 +128,112 @@ contract PluginSetupProcessor is DaoAuthorizable {
     /// @param sender The sender that prepared the plugin installation.
     /// @param dao The address of the dao to which the plugin belongs.
-    /// @param pluginSetup The address of the `PluginSetup` contract used for the update.
-    /// @param data The `bytes` encoded data containing the input parameters for the installation as specified in the `prepareInstallationDataABI()` function in the `pluginSetup` setup contract.
-    /// @param plugin The address of the plugin contract.
-    /// @param updatedHelpers The address array of all helpers (contracts or EOAs) that were prepared for the plugin update.
-    /// @param permissions The list of multi-targeted permission operations to be applied to the installing DAO.
+    /// @param pluginSetupRepo The repository storing the `PluginSetup` contracts of all versions of a plugin.
+    /// @param versionTag The version tag of the plugin to used for install preparation.
+    /// @param setupPayload TOD:GIORGI
+    /// @param preparedDependency TOD:GIORGI
     /// @param initData The initialization data to be passed to the upgradeable plugin contract.
     event UpdatePrepared(
         address indexed sender,
         address indexed dao,
-        address indexed pluginSetup,
-        bytes data,
-        address plugin,
-        address[] updatedHelpers,
-        PermissionLib.ItemMultiTarget[] permissions,
+        bytes32 setupId,
+        PluginRepo indexed pluginSetupRepo,
+        PluginRepo.Tag versionTag,
+        IPluginSetup.SetupPayload setupPayload,
+        IPluginSetup.PreparedDependency preparedDependency,
         bytes initData
     );
 
     /// @notice Emitted after a plugin update was applied.
     /// @param dao The address of the dao to which the plugin belongs.
     /// @param plugin The address of the plugin contract.
-    event UpdateApplied(address indexed dao, address indexed plugin);
+    /// @param setupId TOD:GIORGI: The setup Id hash of the plugin's dependencies.
+    event UpdateApplied(address indexed dao, address indexed plugin, bytes32 setupId);
 
     /// @notice Emitted with a prepared plugin uninstallation to store data relevant for the application step.
     /// @param sender The sender that prepared the plugin uninstallation.
     /// @param dao The address of the dao to which the plugin belongs.
-    /// @param pluginSetup The address of the `PluginSetup` contract used for the uninstallation.
-    /// @param data The `bytes` encoded data containing the input parameters for the uninstallation as specified in the `prepareUninstallationDataABI()` function in the `pluginSetup` setup contract.
-    /// @param plugin The address of the plugin contract.
-    /// @param currentHelpers The address array of all helpers (contracts or EOAs) that were prepared for the plugin to be installed.
+    /// @param setupId TOD:GIORGI: The setup Id hash of the plugin's dependencies.
```

Comment:
Wouldn't it be faster if @novaknole as the author does a rough documentation on his own to then so that I can  polish it? Figuring things out myself might take longer in the end.